### PR TITLE
opt: separator_len works on raw bytes

### PR DESCRIPTION
**Pull request by gabriele-0201**
average 40% improvement from `cargo bench --features benchmarks`

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/465?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#465** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/465?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#464** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/464?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#463** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/463?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#462** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/462?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gabriele-0201 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nomt/src/beatree/ops/bit_ops.rs [hunk 1]
```diff
@@ -456,4 +466,23 @@ mod tests {
             assert_eq!(expected_res, res);
         }
     }
+
+    #[test]
+    fn test_separator_len() {
```
Comment (rphmeier) [inline nomt/src/beatree/ops/bit_ops.rs:471]: style: here and elsewhere, naming tests `test_` is redundant

**Review by rphmeier — COMMENTED** [review]
This also looks fine, and it's great to see the improvement with only a simple change.

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
### Merge activity

* **Oct 13, 2:53 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/465).

File: nomt/src/beatree/ops/bit_ops.rs (hunk 1 above)
Comment (gabriele-0201) [reply nomt/src/beatree/ops/bit_ops.rs:471]: solved in #466

**Pull request merged**